src/app/admin/components/DataTableSectionUser.tsx

```diff
@@ -0,0 +1,381 @@
+import Image from "next/image";
+import { useEffect, useMemo, useState } from "react";
+import { Controller, type SubmitHandler, useForm } from "react-hook-form";
+import { Bounce, toast } from "react-toastify";
+
+import { type Schema } from "@/amplify/data/resource";
+import PopupUser from "@/app/admin/components/PopupTileUser";
+
+const SEARCH_RESULTS_SECTION_STYLES =
+  "bg-white rounded-t-md flex justify-between items-center px-4 py-2 relative";
+const SEARCH_RESULTS_TEXT_STYLES = "py-6 text-2xl font-semibold";
+const SEARCH_BAR_STYLES =
+  "border rounded-md border-black p-4 font-light h-3/5 w-2/5 max-w-[500px]";
+const SEARCH_ICON_STYLES = "absolute right-8 top-1/2 -translate-y-1/2";
+
+const DATA_TABLE_SECTION_STYLES =
+  "bg-light-grey border border-awesomer-purple m-4 p-4 rounded-md text-lg text-black w-full max-w-[1500px]";
+const DATA_TABLE_CONTENT_STYLES =
+  "w-full border-separate border-spacing-x-0.5 text-left";
+const DATA_TABLE_HEADER_CELL_STYLES =
+  "p-3 font-normal text-xl bg-[#A689FF] flex justify-center items-center";
+const DATA_TABLE_CELL_STYLES =
+  "p-3 py-4 text-md font-light overflow-x-auto flex items-center justify-start border-r border-gray-300";
+
+const EDIT_BUTTON_STYLES = "mr-6 text-awesomer-purple";
+const EDIT_MODE_TEXT_INPUT_STYLES =
+  "w-full rounded-md border border-awesomer-purple bg-white p-2 focus:outline-none focus:ring-1 focus:ring-awesomer-purple";
+const CHANGE_PAGE_BUTTON_STYLING =
+  "rounded-md border border-awesomer-purple bg-white px-6 hover:bg-awesomer-purple hover:text-white";
+const CHANGE_PAGE_BUTTON_TEXT_STYLING = "rounded-md bg-white p-2 px-8";
+
+const search_icon = "/svgs/admin/search_icon.svg";
+
+const entries_per_page = 10;
+
+interface DataTableProps {
+  tableHeaders: Array<{ columnHeader: string }>;
+  userData: Array<Partial<Schema["User"]["type"]>>;
+  tableDataMutation: any;
+}
+
+const DataTableSectionUser = (props: DataTableProps) => {
+  const { control, handleSubmit, reset } = useForm<Schema["User"]["type"]>();
+
+  const onSubmit: SubmitHandler<Schema["User"]["type"]> = (
```

Comment:
this method is using setEditingId before it is declared, please reorder the methods